lib/concurrent-ruby/concurrent/utility/processor_counter.rb

```diff
@@ -112,7 +112,9 @@ def compute_cpu_quota
           elsif File.exist?("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us")
             # cgroups v1: https://kernel.googlesource.com/pub/scm/linux/kernel/git/glommer/memcg/+/cpu_stat/Documentation/cgroups/cpu.txt
             max = File.read("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us").to_i
-            return nil if max == 0
+            # If the cpu.cfs_quota_us is -1, cgroup does not adhere to any CPU time restrictions
+            # https://docs.kernel.org/scheduler/sched-bwc.html#management
+            return nil if max == 0 || max == -1
```

Comment:
Maybe we should check `if max <= 0`?
Seems a tiny bit safer and more efficient as we can't interpret a value <= 0 safely.
Could you change to that?

Reply:
Thanks for your review. I updated the condition.